### ISSUE TITLE
PyMVPA aborts when loading fmri dataset

### ISSUE THREAD

**Issue by samfux84** [PyMVPA/PyMVPA#558]
I am running a Python code using pyMVPA and when using the function

db = mvpa2.datasets.mri.fmri_dataset(
                                     nii_fns, targets=labels, chunks=grps, mask='/cluster/work/wenderoth/PKF/SearchLight/Resources/maskTesting.nii', sprefix=sprefix_, tprefix=tprefix_, add_fa=None)

it aborts with

```
loading data
Traceback (most recent call last):
  File "SL_masked.py", line 71, in <module>
    nii_fns, targets=labels, chunks=grps, mask='/cluster/work/wenderoth/PKF/SearchLight/Resources/maskTesting.nii', sprefix=sprefix_, tprefix=tprefix_, add_fa=None)
  File "/cluster/apps/python/3.6.0/x86_64/lib64/python3.6/site-packages/mvpa2/datasets/mri.py", line 254, in fmri_dataset
    imgdata, imghdr, img = _load_anyimg(samples, ensure=True, enforce_dim=4)
TypeError: 'NoneType' object is not iterable
-bash-4.2$
```

Any idea what could cause such a problem ?

**Comment by yarikoptic** [comment]
ok, python3.6 ... unfortunately I must say that we aren't yet fully ready for python3 and various issues might emerge 

anyways -- are you sure that `nii_fns` isn't an empty list or smth?

**Comment by samfux84** [comment]
@yarikoptic Thank you for your reply. I will try to install mvpa2 for a Python 2.7.x version and test if this solves the problem. As soon as I could test this I will provide you some feedback.

**Comment by yarikoptic** [comment]
note that we have pymvpa2 within conda-forge now, so it might be the easiest way for you to install it on your non-debian system

**Comment by samfux84** [comment]
@yarikoptic Thank you for the hint about conda-forge, but I prefer to install all software directly from source.

After having installed mvpa2 for Python 2.7.14, I figured out that the paths to the data files were wrong.

I corrected the paths and now it runs fine with both versions (Python 2.7.14 and Python 3.6.0).

Thank you very much for your help.

**Comment by yarikoptic** [comment]
"runs fine" is great, but I would strongly advise to use python2 for now -- functioning on python3 is undertested to say the least.  So even if it would "run fine" (i.e. not crash) it might produce incorrect results.  Just try doing `python -m nose -s -v mvpa2` or `mvpa2.test()` (withing python3) to see what aspects are known to be broken

**Comment by samfux84** [comment]
@yarikoptic Thank you for your reply.

I have installed mvpy2 on our cluster because one of our users requested it. The user provided me a test case that now runs through without crashing, therefore I was writing in my last reply that it "runs fine".

Thank you for your heads up about Python 3.

I have forwarded your comment to the users and recommended based on your statement to only use the Python 2.7.14 installation of mvpa2.

Just in case you are interested in the results of the tests that I have run for the Python 3.6.0 installation with the command that you provided (python -m nose -s -v mvpa2):

[test.txt](https://github.com/PyMVPA/PyMVPA/files/1521845/test.txt)

```
Ran 548 tests in 200.784s

FAILED (SKIP=39, errors=19, failures=9)
```
8 Tests failed because I did not have an X11 connection to the HPC cluster. They all failed with

`_tkinter.TclError: no display name and no $DISPLAY environment variable`

Best regards
